docs/release_notes.md

```diff
@@ -1,179 +0,0 @@
-# Release Notes
```

Comment:
Should we simply start [automatically generating release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)?

Reply:
That's what we switched to for the last couple releases, which is why I'm just getting rid of this file now.